### DISCUSSION TITLE
Close Session Manually from the Server

### DISCUSSION THREAD

**Discussion by Charlyx12** [FreeOpcUa/opcua-asyncio#854] [Q&A] [answered]
Hi! How could I close the session of an active user manually from the server? For example, if the password of the client has expired I want the server to eject him with no needing of the client to send the _CloseSessionRequest_. 

I´m trying to use the `close_session` method from _internal_session.py_ in this way:
`server.bserver.clients[0].processor.session.close_session(True) `
Where server.bserver.clients is the list that contains the info about the connected clients. With the index 0 im referring to the first client. This doesn´t work. Some idea?

Thank you

**Comment by schroeder-** [comment, marked as answer]
This should work:

```
await server.bserver.clients[0].processor.session.close_session(True)
server.bserver.clients[0].close()
```

**Reply by Charlyx12** [reply]
Thanks for the answer.
I tried it but gives an error saying the method close() does not belong to the object OPCUAProtocol. I changed _close()_ to _connection_lost()_, which is an existing function from OPCUAProtocol class and it already does a _close()_. 
To sum up, this looks to work:
`await server.bserver.clients[0].processor.session.close_session(True)`
`server.bserver.clients[0].connection_lost("a")`
_connection_lost()_ requires a parameter that will be used in a print function so i dont really care about it and i put a random value.
